jupyter_book/build.py

```diff
@@ -97,69 +95,58 @@ def _case_sensitive_fs(path):
-    # Default values and arguments
-
-    args = parser.parse_args(sys.argv[2:])
-    overwrite = bool(args.overwrite)
-    execute = bool(args.execute)
-
-    # Paths for our notebooks
-    PATH_BOOK = op.abspath(args.path_book)
-
-    PATH_TOC_YAML = args.toc if args.toc is not None else op.join(
-        PATH_BOOK, '_data', 'toc.yml')
-    CONFIG_FILE = args.config if args.config is not None else op.join(
-        PATH_BOOK, '_config.yml')
-    PATH_TEMPLATE = args.template if args.template is not None else op.join(
-        PATH_BOOK, 'scripts', 'templates', 'jekyllmd.tpl')
-    PATH_IMAGES_FOLDER = op.join(PATH_BOOK, '_build', 'images')
-    BUILD_FOLDER = op.join(PATH_BOOK, BUILD_FOLDER_NAME)
-
-    ###############################################################################
+def build_book(path_book, path_toc_yaml=None, config_file=None,
+               path_template=None, local_build=False, execute=False,
+               overwrite=False):
+    """Build the markdown for a book using its TOC and a content folder.
+
+    Parameters
+    ----------
+    path_book : str
+        Path to the root of the book repository
+    path_toc_yaml : str | None
+        Path to the Table of Contents YAML file
+    config_file : str | None
+        Path to the Jekyll configuration file
+    path_template : str | None
+        Path to the template nbconvert uses to build markdown
+        files
+    local_build : bool
+        Specify you are building site locally for later upload
+    execute : bool
+        Whether to execute notebooks before converting to markdown
+    overwrite : bool
+        Whether to overwrite existing markdown files
+    """
+
+    PATH_IMAGES_FOLDER = op.join(path_book, '_build', 'images')
+    BUILD_FOLDER = op.join(path_book, BUILD_FOLDER_NAME)
+
+    ###############################################
     # Read in textbook configuration
 
     # Load the yaml for this site
-    with open(CONFIG_FILE, 'r') as ff:
+    with open(config_file, 'r') as ff:
         site_yaml = yaml.load(ff.read())
     CONTENT_FOLDER_NAME = site_yaml.get('content_folder_name').strip('/')
-    PATH_CONTENT_FOLDER = op.join(PATH_BOOK, CONTENT_FOLDER_NAME)
+    PATH_CONTENT_FOLDER = op.join(path_book, CONTENT_FOLDER_NAME)
 
     # Load the textbook yaml for this site
-    if not op.exists(PATH_TOC_YAML):
+    if not op.exists(path_toc_yaml):
         raise _error(
-            "No toc.yml file found, please create one at `{}`".format(PATH_TOC_YAML))
-    with open(PATH_TOC_YAML, 'r') as ff:
+            "No toc.yml file found, please create one at `{}`".format(path_toc_yaml))
+    with open(path_toc_yaml, 'r') as ff:
         toc = yaml.load(ff.read())
 
     # Drop divider items and non-linked pages in the sidebar, un-nest sections
     toc = _prepare_toc(toc)
 
-    ###############################################################################
+    ################################################
```

Comment:
@choldgraf just trying to slowly make it more pep8 compliant :) So we can remove the ignore for the 80 line rule eventually ...